ansible/roles/providers/tasks/aws.yml

```diff
@@ -33,6 +33,36 @@
   when:
     - ansible_distribution == "Flatcar"
 
+
+- name: install cloud-init deps
+  yum:
+    name: "{{ packages }}"
+    state: present
+    enablerepo: "{{ 'offline' if offline_mode_enabled else '' }}"
+    disablerepo: "{{ '*' if offline_mode_enabled else '' }}"
+  vars:
+    packages:
+      - cloud-utils-growpart
+      - "python{{ '3' if ansible_distribution_major_version|int >= 9 else '2'}}-pip"
+  when: ansible_os_family == "RedHat"
+
+- name: install cloud-init for RHEL
+  yum:
+    name: cloud-init
+    state: present
+    enablerepo: "{{ 'offline' if offline_mode_enabled else '' }}"
+    disablerepo: "{{ '*' if offline_mode_enabled else '' }}"
+  when:
+  - ansible_os_family == "RedHat"
+  - ansible_distribution_version is version('8.10', '!=')
+
+ #this pins the version of cloud-init to avoid the issue here https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4745
+- name: Install cloud-init 23.1.1 for RHEL 8.10
+  shell: rpm -Uvh --oldpackage "{{ offline.os_packages_remote_filesystem_repo_path }}/cloud-init-23.1.1-10.el8.noarch.rpm"
+  when:
+  - ansible_os_family == "RedHat"
```

Comment:
we should check for `offline_mode_enabled` if this task takes rpm from the offline path.
We need to handle `networked` usecase too.

Reply:
> We need to handle networked usecase too.

Wdym?

Reply:
wont `rpm -Uvh --oldpackage "{{ offline.os_packages_remote_filesystem_repo_path }}/cloud-init-23.1.1-10.el8.noarch.rpm"` fail in `basic` installation?

Or it is handled somewhere else?

Reply:
addressed in the next commit i think